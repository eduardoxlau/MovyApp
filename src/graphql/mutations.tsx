import { gql } from '@apollo/client';

export const LOGIN = gql`
  mutation Login($input: LoginInput!) {
    login(input: $input) {
      access_token
      user {
        id
        email
        full_name
        admin
        photo_path
      }
    }
  }
`;
export const CREATE_USER = gql`
  mutation CreateUser($input: CreateUserInput!) {
    createUser(input: $input) {
      id
      email
      full_name
      admin
      photo_path
    }
  }
`;

export const UPDATE_USER = gql`
  mutation UpdateUser($input: UpdateUserInput!) {
    updateUser(input: $input) {
      status
    }
  }
`;

export const REMOVE_USER = gql`
  mutation DeleteUser {
    deleteUser {
      status
    }
  }
`;

export const CREATE_LIST = gql`
  mutation CreateList($input: CreateListInput!) {
    createList(input: $input) {
      id
      name
      description
    }
  }
`;

export const EDIT_LIST = gql`
  mutation UpdateList($input: UpdateListInput!) {
    updateList(input: $input) {
      status
    }
  }
`;

export const REMOVE_LIST = gql`
  mutation DeleteList($id: Float!) {
    deleteList(id: $id) {
      status
    }
  }
`;

export const REMOVE_MOVIE_LIST = gql`
  mutation RemoveMovieToList($input: ListMovieInput!) {
    removeMovieToList(input: $input) {
      status
    }
  }
`;

export const ADD_MOVIE_LIST = gql`
  mutation AddMovieToList($input: ListMovieInput!) {
    addMovieToList(input: $input) {
      status
    }
  }
`;

export const SEEN_MOVIE = gql`
  mutation seenMovie($movieId: Float!) {
    seenMovie(movieId: $movieId) {
      status
    }
  }
`;
