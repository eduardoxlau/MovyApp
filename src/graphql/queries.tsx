import { gql } from '@apollo/client';

export const GET_USER = gql`
  query GetUser {
    getUser {
      id
      email
      full_name
      admin
      lists {
        description
        movies {
          id
          title
        }
      }
    }
  }
`;

export const GET_MOVIES_SEEN = gql`
  query GetMoviesSeen {
    getMoviesSeen {
      id
      title
      overview
      trailer_url
      card_url
      poster_url
      backdrop_url
      stars
      genres {
        name
      }
    }
  }
`;

export const GET_MOVIE = gql`
  query GetMovie($id: Float!) {
    getMovie(id: $id) {
      id
      title
      overview
      trailer_url
      card_url
      poster_url
      backdrop_url
      stars
      genres {
        name
      }
    }
  }
`;

export const GET_MOVIES = gql`
  query GetMovies {
    getMovies(limit: 20) {
      items {
        id
        title
        overview
        trailer_url
        card_url
        poster_url
        backdrop_url
        stars
        genres {
          name
        }
      }
    }
  }
`;

export const GET_MOVIES_PAGINATE = gql`
  query GetMovies($page: Float, $limit: Float) {
    getMovies(page: $page, limit: $limit) {
      meta {
        totalPages
        currentPage
      }
      items {
        id
        title
        overview
        trailer_url
        card_url
        poster_url
        backdrop_url
        stars
        genres {
          name
        }
      }
    }
  }
`;

export const GET_LISTS = gql`
  query GetLists {
    getLists {
      id
      name
      description
      movies {
        id
        title
        trailer_url
        card_url
      }
    }
  }
`;
