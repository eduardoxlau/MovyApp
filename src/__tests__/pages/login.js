/* eslint-disable import/no-extraneous-dependencies */
/* eslint-disable no-undef */
import wait from 'waait';
import renderer from 'react-test-renderer';
import { render, fireEvent } from '@testing-library/react';
import { MockedProvider } from '@apollo/react-testing';
import { createMemoryHistory } from 'history';
import { Router } from 'react-router-dom';

import { LOGIN } from 'graphql/mutations';
import Login from 'views/login';

const user = { email: 'test@gmail', password: 'password' };

const mocks = [
  {
    request: {
      query: LOGIN,
      variables: { input: user },
    },
    result: { data: { login: { access_token: 'XXXXXX', user } } },
  },
];

describe('<Login />', () => {
  const history = createMemoryHistory();
  it('should render snapshot Login', () => {
    const tree = renderer
      .create(
        <MockedProvider mocks={mocks}>
          <Login />
        </MockedProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should submit form login', async () => {
    const { getByText } = render(
      <Router history={history}>
        <MockedProvider mocks={mocks} addTypename={false}>
          <Login />
        </MockedProvider>
      </Router>
    );
    fireEvent.change(document.querySelector('input[name="email"]'), {
      target: { value: user.email },
    });

    fireEvent.change(document.querySelector('input[name="password"]'), {
      target: { value: user.password },
    });
    fireEvent.click(getByText('Iniciar sesi√≥n'));
    await wait(200);
    await expect(history.length).toBe(2);
  });
});
