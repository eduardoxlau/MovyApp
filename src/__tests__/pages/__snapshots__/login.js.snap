// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> should render snapshot Login 1`] = `
<div
  className="flex flex-col bg-logo bg-cover"
>
  <form
    onSubmit={[Function]}
  >
    <div
      className="container flex-grow flex-shrink mx-auto mt-44"
    >
      <div
        className="bg-black text-white m-auto px-12 py-12 w-auto md:w-login bg-opacity-80 rounded"
      >
        <div
          className="text-4xl text-center"
        >
          Inicia sesión
        </div>
        <div
          className="my-9"
        >
          <input
            className="rounded w-full h-10 my-1.5 text-black indent-1"
            name="email"
            onChange={[Function]}
            placeholder="Email o número de teléfono"
            required={true}
            type="email"
            value=""
          />
          <input
            className="rounded w-full h-10 my-1.5 text-black indent-1"
            name="password"
            onChange={[Function]}
            placeholder="Contraseña"
            required={true}
            type="password"
            value=""
          />
        </div>
        <input
          className="w-full h-11 text-center rounded cursor-pointer bg-button"
          disabled={false}
          type="submit"
          value="Iniciar sesión"
        />
        <div
          className="flex items-center my-3"
        >
          <div
            className="w-1/2 flex"
          >
            <div
              className="flex items-center"
            >
              <input
                className="mr-5 w-5 h-5 bg-coolGray-400"
                type="checkbox"
              />
            </div>
            Recuerdame
          </div>
          <div
            className="w-1/2 flex justify-end"
          >
            Necesitas ayuda?
          </div>
        </div>
        <div
          className="my-10 flex justify-center items-center"
        >
          <img
            alt=""
            className="mr-4"
            src="fb.png"
          />
          Iniciar sesión con Facebook
        </div>
        <div
          className="text-center"
        >
          ¿Primera vez en Movy?
          <span
            className="font-bold"
          >
            Suscríbete ya.
          </span>
        </div>
      </div>
    </div>
    <div
      className="h-40 bglinear-black"
    />
  </form>
</div>
`;
